lib/doorkeeper/models/access_token_mixin.rb

```diff
@@ -128,7 +128,16 @@ def generate_refresh_token
     end
 
     def generate_token
-      self.token = UniqueToken.generate
+      generator = Doorkeeper.configuration.access_token_generator.constantize
+
+      if generator.method(:generate).arity == 1
+        self.token = generator.generate(resource_owner_id)
+      else
+        self.token = generator.generate
+      end
+    rescue ArgumentError, TypeError, NoMethodError, NameError
```

Comment:
I'd trap `NoMethodError` as we are doing it, and `NameError` in another block with a proper error message (something like "generator not found"). I don't think we'll need to trap the other two?